openedx/features/course_experience/views/course_home.py

```diff
@@ -121,7 +124,9 @@ def render_to_fragment(self, request, course_id=None, **kwargs):
         course = get_course_with_access(request.user, 'load', course_key)
 
         # Render the course dates as a fragment
-        dates_fragment = CourseDatesFragmentView().render_to_fragment(request, course_id=course_id, **kwargs)
+        course_expiration_message = generate_course_expired_message(request.user, course)
```

Comment:
So I made this switch because I had a failing test for query counts (it went up by 1). Not sure if this is better or to just accept the extra query and leave this logic inside of the `CourseDatesFragmentView().render_to_fragment` method. (see commit 1 for the difference)